corelibs/libgraph.md

```diff
@@ -1,36 +1,352 @@
+
+    - for `16-bit` color depth - it's the following format in bits: `RRRRRGGGGGGBBBBB`, so `0x07E0` represents green.
+
+    - for `24-bit` color depth - it's represented by `0xRRGGBB` so `0x0000FF` is blue.
+
+    - for `32-bit` color depth - it's the following format in hex: `0xAARRGGBB`, where `A` represents alpha. With `AA` set to `0xFF` opacity equals 100%, but by default transparency isn't enabled so there can be any value.
+
+  - _`queue`_ - graphics task queue, should be chosen from the `graph_queue_t` enum, where are following options: `GRAPH_QUEUE_HIGH` - high priority queue, `GRAPH_QUEUE_LOW` - low priority queue, `GRAPH_QUEUE_BOTH` - both queues are chosen, `GRAPH_QUEUE_DEFAULT` - default queue.
+
+
+- `int graph_rect(graph_t *graph, unsigned int x, unsigned int y, unsigned int dx, unsigned int dy, unsigned int color, graph_queue_t queue)` </br>
+  Draws a rectangle. Arguments are similar to those in graph_line function. A drawn rectangle will be filled with a specified color.
+
+- `int graph_fill(graph_t *graph, unsigned int x, unsigned int y, unsigned int color, graph_fill_t type, graph_queue_t queue)` </br>
+  Fills polygon limited by drawn lines (when type is `GRAPH_FILL_FLOOD`), or limited by screen boundaries (`GRAPH_FILL_BOUND` type).
+
+- `int graph_print(graph_t *graph, const graph_font_t *font, const char *text, unsigned int x, unsigned int y, unsigned char dx, unsigned char dy, unsigned int color, graph_queue_t queue)` </br>
+  Prints text pointed by the _`text`_ argument. Font data should be passed to `graph_font_t` structure. The example is stored in `gfx` directory in [phoenix-rtos-tests](https://github.com/phoenix-rtos/phoenix-rtos-tests.git) repository (`font.h` file). The remaining arguments are similar to those from functions above.
+
+- `int graph_move(graph_t *graph, unsigned int x, unsigned int y, unsigned int dx, unsigned int dy, int mx, int my, graph_queue_t queue)` </br>
+  Moves data in the range specified by _`x`_, _`y`_, _`dx`_, _`dy`_ arguments to the following point: (_`x`_ + _`mx`_, _`y`_ + _`my`_).
+
+- `graph_copy(graph_t *graph, const void *src, void *dst, unsigned int dx, unsigned int dy, unsigned int srcspan, unsigned int dstspan, graph_queue_t queue)` </br>
+  Copies a bitmap pointed by the _`src`_ argument into bitmap pointed by the _`dst`_ argument. The area, which is copied is limited by a rectangle with _`dx`_ and _`dy`_ dimensions. There should also be specified span arguments, which is the total width of a source/destination bitmap multiplied by a color depth. When copying some part of a bitmap, _`src`_ should point to the proper element, same with destination buffer. By default, the following bitmap's format is applied: {B, G, R, A, B, G, R, A}, so when color depth equals 4 in one row of array are stored 2 pixels. For cirrus graphics adapter (default), alpha (represented by `A`) isn't supported, so opacity is always 100%.
```

Comment:
> By default, the following bitmap's format is applied: {B, G, R, A, B, G, R, A}, so when color depth equals 4 in one row of array are stored 2 pixels.

I'm not sure what that means. The bitmap format doesn't really matter - we're just copying data. It's up to the function caller to keep track of the data structure.